### PR TITLE
v2.7.0 - The biggest single update yet!

### PR DESCRIPTION
**Pull request by UncleClapton**
## Additions

* Front page slideshow/carousel
    * With image loading web worker logic by @EladKarni
* New rat manager so you can create, rename, and delete rats at will. (ship manager coming soon!)
* Rescue list/search page for overseer+
* Global state selectors to simplify redux state mapping.
* Page url parameter for profile page tabs.

## Changes

* Redesigned paperwork view page. 
    * This is also @314numberpi's first major contribution as our latest tech rat! Welcome to the team.
* Redesign store products page slightly so it's nicer to interact with. 
* Add rescue header and quotes to paperwork edit form.
* Add simple page transitions. Now you're playing with (`react-spring`) power!
* Improved switch toggle by adding icons to better illustrate state
    * This has the added effect of also making the switch color-blind friendly
* Add tooltips/error messages to ValidatedFormInput - Thanks @EladKarni
* Display TOS/COC pages after the user attempts to submit registration form - Thanks @EladKarni!
* Automatically trim whitespace for IRC nicks and CMDR names - Thanks @EladKarni!
* Upgrade to Next 8
* Update TabbedPanel tab appearance
* removed an easter egg. (but it's okay since it was removed to make way for a real implementation of what the easter egg did.)
* unmount page from app if user logs out while on authenticated page.
* Better project documentation
* Update project code style

## Fixes

* Fix up SCSS styles to comply with node-sass
* Fix store order listing page breaking when products/skus no longer exist
* Various other small bugfixes and overall improvments

### CODE REVIEW

File: client/components/AddRatForm.js [hunk 1]
```diff
@@ -6,12 +6,33 @@ import React from 'react'
...
+import ValidatedFormInput from './ValidatedFormInput'
+import ValidatedFormSelect from './ValidatedFormSelect'
 
 
 
 
+// Component Constants
+const INVALID_NAME_MESSAGE = 'CMDR Name is Required'
+const INVALID_PLATFORM_MESSAGE = 'Platform is Required'
+const initialState = {
```
Comment (theunkn0wn1) [inline client/components/AddRatForm.js:23]: name does not conform to constants naming scheme, shouldn't this be `INITIAL_STATE`?

File: client/components/AddRatForm.js [hunk 2]
```diff
@@ -47,9 +66,42 @@ class AddRatForm extends Component {
...
+    }))
+  }
+
+  _handleFieldChange = ({ target, valid, message }) => {
+    this.setState(({ validity }) => {
+      const {
+        name,
+        value,
+      } = target
+      const required = typeof validity[name] !== 'undefined'
```
Comment (theunkn0wn1) [inline client/components/AddRatForm.js:91]: naming scheme, shouldn't this be `REQUIRED`?

**Review by theunkn0wn1 — COMMENTED** [review]
there probably are a few more constants misnamed, though il stop here and ask if they were intentionally not matching style

**Comment by UncleClapton** [comment]
Found a bug which essentially crashes the website when you logout on a page requiring authentication. Fixing before release.

**Comment by UncleClapton** [comment]
No other bugs found. Ready for release.

**Pull request merged**